### PR TITLE
feat: ux-layout

### LINKED ISSUE #42: Ajouter dossiers et dates dans la visualisation générale
### EPIC

https://github.com/SocialGouv/archifiltre-private/issues/3

### AFTER

#27

### Contexte

Idem US https://github.com/SocialGouv/archimail/issues/27

### Définition du(des) problème(s)

Idem US https://github.com/SocialGouv/archimail/issues/27

### Solution

ETQ utilisateur d'Archimail
JVP visualiser dans la vue générale:
- un logo et le nombre de dossiers présents dans la messagerie
- les dates extrêmes des éléments contenus dans la messagerie (la date du mail le + ancien et la date du mail le + récent sous la forme JJ/MM/AAAA-JJ/MM/AAAA)

AD comprendre le contenu de la messagerie

### Critères d'acceptance

_No response_

### Les tests e2e doivent-ils passer ?

- [ ] e2e

### Valeur Business

BV:M

### LINKED ISSUE #45: Créer l'interface d'Archimail
### EPIC

https://github.com/SocialGouv/archifiltre-private/issues/3

### AFTER

#

### Contexte

Archimail est un nouvel outil développé pour la visualisation, l'appréhension, le traitement et la recherche dans les messageries.

### Définition du(des) problème(s)

Pour ces différentes fonctionnalités, nous avons besoin d'une nouvelle interface adaptée au contexte des messageries qui s'approche du design d'Archifiltre .

### Solution

ETQ utilisateur d'Archimail
JVP visualiser, appréhender, traiter et rechercher à partir d'une interface comprenant:
- 1 barre de boutons d'actions
- 1 partie data-viz
- 1 bandeau latéral avec plusieurs blocs à afficher: grands ensembles de la messagerie, informations sur 1 élément, impacts à l'élimination

AD pouvoir utiliser l'outil
cf. maquette Figma : https://www.figma.com/file/3MtupS5WM82WIWUCOLrnys/Archifiltre?node-id=0%3A1

### Critères d'acceptance

_No response_

### Les tests e2e doivent-ils passer ?

- [ ] e2e

### Valeur Business

BV:XL

### PR DESCRIPTION
**Pull request by mehdilouraoui**
Closes #45
Closes #42 

# Objectif(s)

### CODE REVIEW

File: src/renderer/views/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -1,63 +1,26 @@
...
-                title: "Agenda",
-            },
-        ],
-        labels: ["Catégorie", "Pourcentage", "Mail", "Pièces Jointes"],
-        title: "Agenda",
-    },
-];
-
-export const Dashboard: React.FC = () => {
+export const Dashboard: FC = () => {
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/Dashboard.tsx:11]: Why use `FC` as is if you already have `React` object usable?
```suggestion
export const Dashboard: React.FC = () => {
```

File: src/renderer/views/dashboard/DashboardActions.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import type { FC } from "react";
+import React, { useCallback, useState } from "react";
+
+import {
+    ExportPicto,
+    FilterPicto,
+    ImportPicto,
+    SearchPicto,
+} from "../../../renderer/components/common/pictos/picto";
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardActions.tsx:9]:
```suggestion
} from "../../components/common/pictos/picto";
```

File: src/renderer/views/dashboard/DashboardInformations.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable react/no-unescaped-entities */
+import type { FC } from "react";
+import React from "react";
+
+import { Card } from "../../../renderer/components/common/card/Card";
+import { usePSTStore } from "../../../renderer/store/PSTStore";
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardInformations.tsx:6]:
```suggestion
import { Card } from "../../components/common/card/Card";
import { usePstStore } from "../../store/PstStore";
```

File: src/renderer/views/dashboard/DashboardInformations.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                            {JSON.stringify(mainInfos.sentTime)}
+                        </li>
+                        <li>
+                            <span>Date de réception : </span>
+                            {JSON.stringify(mainInfos.receivedDate)}
+                        </li>
+                        <li>
+                            <span>Adresse mail : </span>
+                            {JSON.stringify(mainInfos.from, null, 1)}
+                            {/* {mainInfos.from.email ?? null} */}
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardInformations.tsx:41]: delete

File: src/renderer/views/dashboard/DashboardInformations.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                        </li>
+                    </ul>
+                ) : (
+                    <p>
+                        Passer le curseur sur le visualiseur pour afficher des
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardInformations.tsx:46]: We must not forget to add a TODO for i18n on all "words"

File: src/renderer/views/dashboard/DashboardRecap.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+import { usePSTStore } from "../../../renderer/store/PSTStore";
+import {
+    getPstMailsPercentage,
+    getPstTotalContacts,
+    getPstTotalDeletedMails,
+    getPstTotalReceivedAttachments,
+    getPstTotalReceivedMails,
+    getPstTotalSentAttachments,
+    getPstTotalSentMails,
+} from "../../../renderer/utils/pst-extractor";
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardRecap.tsx:20]:
```suggestion
import { Card } from "../../components/common/card/Card";
import {
    ContactPicto,
    MailPicto,
    MailSentPicto,
    TrashPicto,
} from "../../components/common/pictos/picto";
import { usePstStore } from "../../store/PstStore";
import {
    getPstMailsPercentage,
    getPstTotalContacts,
    getPstTotalDeletedMails,
    getPstTotalReceivedAttachments,
    getPstTotalReceivedMails,
    getPstTotalSentAttachments,
    getPstTotalSentMails,
} from "../../utils/pst-extractor";
```

File: src/renderer/views/dashboard/DashboardRecapSelectFolder.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import type { FC } from "react";
+import React from "react";
+
+import { usePSTFolderList } from "../../hooks/usePSTFolderList";
+import style from "./Dashboard.module.scss";
+import { DashboardRecapSelectFolderItem } from "./DashboardRecapSelectFolderItem";
+
+interface DashboardRecapSelectFolderProps {
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardRecapSelectFolder.tsx:8]: Always export props
```suggestion
export interface DashboardRecapSelectFolderProps {
```

File: src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import type { FC } from "react";
+import React from "react";
+
+import { usePSTStore } from "../../../renderer/store/PSTStore";
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx:4]:
```suggestion
import { usePstStore } from "../../store/PstStore";
```

File: src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import style from "./Dashboard.module.scss";
+
+interface DashboardRecapSelectFolderItemProps {
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx:7]:
```suggestion
export interface DashboardRecapSelectFolderItemProps {
```

File: src/renderer/views/start-screen/StartScreen.tsx [hunk 10]
```diff
@@ -1,28 +1,66 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useEffect } from "react";
 
 import { useRouteContext } from "../../../renderer/context/RouterContext";
-import { LayoutWorkspace } from "../../components/common/layout/LayoutWorkspace";
+import { usePSTExtractor } from "../../../renderer/hooks/usePSTExtractor";
```
Comment (lsagetlethias) [inline src/renderer/views/start-screen/StartScreen.tsx:4]:
```suggestion
import { usePSTExtractor } from "../../hooks/usePSTExtractor";
```

File: src/renderer/views/start-screen/StartScreen.tsx [hunk 11]
```diff
@@ -1,28 +1,66 @@
...
+        [setPstFilePath]
     );
+
+    useEffect(() => {
+        if (pstFile) {
+            changeRoute("DASHBOARD");
+        }
+    }, [pstFile, changeRoute]);
+
+    console.log(pstProgress);
```
Comment (lsagetlethias) [inline src/renderer/views/start-screen/StartScreen.tsx:31]: delete

File: src/renderer/views/start-screen/StartScreenLoader.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import type { PstProgressState } from "@common/modules/pst-extractor/type";
+import React from "react";
+
+import style from "./StartScreen.module.scss";
+
+type StartScreenLoaderProps = Pick<
```
Comment (lsagetlethias) [inline src/renderer/views/start-screen/StartScreenLoader.tsx:6]:
```suggestion
export type StartScreenLoaderProps = Pick<
```

File: src/renderer/views/start-screen/StartScreenLoader.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+export const StartScreenLoader: React.FC<StartScreenLoaderProps> = ({
+    countEmail,
+    countFolder,
+    countAttachement,
+    countTotal,
+}) => (
+    <>
+        <div className={style.spinner} />
+        <div className={style["circle-packing-pst-infos"]}>
```
Comment (lsagetlethias) [inline src/renderer/views/start-screen/StartScreenLoader.tsx:19]:
```suggestion
        <div className={style.circlePackingPstInfos}>
```

**Review by lsagetlethias — CHANGES_REQUESTED** [review]

File: src/renderer/views/start-screen/StartScreenLoader.tsx (hunk 12 above)
Comment (mehdilouraoui) [reply src/renderer/views/start-screen/StartScreenLoader.tsx:6]: done

File: src/renderer/views/start-screen/StartScreenLoader.tsx (hunk 13 above)
Comment (mehdilouraoui) [reply src/renderer/views/start-screen/StartScreenLoader.tsx:19]: pas possible avec le kebab case

File: src/renderer/views/start-screen/StartScreen.tsx (hunk 10 above)
Comment (mehdilouraoui) [reply src/renderer/views/start-screen/StartScreen.tsx:4]: done

File: src/renderer/views/start-screen/StartScreen.tsx (hunk 11 above)
Comment (mehdilouraoui) [reply src/renderer/views/start-screen/StartScreen.tsx:31]: done

File: src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx (hunk 9 above)
Comment (mehdilouraoui) [reply src/renderer/views/dashboard/DashboardRecapSelectFolderItem.tsx:7]: done

File: src/renderer/views/dashboard/DashboardRecapSelectFolder.tsx (hunk 7 above)
Comment (mehdilouraoui) [reply src/renderer/views/dashboard/DashboardRecapSelectFolder.tsx:8]: done

File: src/renderer/views/dashboard/DashboardInformations.tsx (hunk 4 above)
Comment (mehdilouraoui) [reply src/renderer/views/dashboard/DashboardInformations.tsx:41]: done

File: src/renderer/views/dashboard/DashboardInformations.tsx (hunk 5 above)
Comment (mehdilouraoui) [reply src/renderer/views/dashboard/DashboardInformations.tsx:46]: yes... should be a solo PR.

File: src/renderer/views/dashboard/Dashboard.tsx (hunk 1 above)
Comment (mehdilouraoui) [reply src/renderer/views/dashboard/Dashboard.tsx:11]: Readability but fair enough to use React.FC

File: src/renderer/utils/pst-extractor.ts [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+
+/**
+ * Get the total sent emails attachements counts of a given PST.
+ */
+export const getPstTotalSentAttachments = (
+    extractTables: PstExtractTables | undefined
+): number => {
+    if (extractTables) {
+        return [...extractTables.emails.values()]
+            .flat(9) // TODO magic number
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/utils/pst-extractor.ts:66]: done

File: src/renderer/utils/pst-extractor.ts [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+    return specificFolder[0]?.children?.length;
+};
+
+/**
+ * Get all folder name from a given PST.
+ */
+export const getPSTRootFolderList = (pst: PstContent): string[] =>
+    pst.children[0]!.children!.map(({ name }) => name);
+
+// domaine > années > mails
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/utils/pst-extractor.ts:96]: done

File: src/renderer/utils/pst-extractor.ts [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+export interface PstComputed {
+    id: string;
+    name: string;
+    size: number;
+    value: string;
+    children: PstComputedChild[];
+}
+
+//TODO: comment
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/utils/pst-extractor.ts:113]: done

File: src/renderer/styles/_font.scss [hunk 17]
```diff
@@ -1,18 +1,18 @@
-@font-face {
-  font-family: "Quicksand";
-  font-style: normal;
-  font-weight: 400;
-  src: url("../../../static/fonts/Quicksand/Quicksand-Regular.ttf")
-    format("opentype");
-}
+// @font-face {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/styles/_font.scss:1]: done

File: src/renderer/store/TagManagerStore.tsx [hunk 18]
```diff
@@ -0,0 +1,40 @@
+import { atom, useAtom } from "jotai/index";
+import type { SetStateAction } from "react";
+import { useCallback } from "react";
+
+export interface UseTagManagerInterface {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/TagManagerStore.tsx:5]: done

File: src/renderer/store/TagManagerStore.tsx [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    hoveredId: string;
+    setHoveredId: (update: SetStateAction<string>) => void;
+    addMarkedToDelete: () => void;
+}
+
+const markedToDeleteAtom = atom<string[]>([""]);
+const markedToKeepAtom = atom<string[]>([""]);
+const hoveredIdAtom = atom<string>("");
+
+export const useTagManagerStore = (): UseTagManagerInterface => {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/TagManagerStore.tsx:19]: done

File: src/renderer/store/PSTStore.tsx [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    PstFolder,
+} from "@common/modules/pst-extractor/type";
+import { atom, useAtom } from "jotai/index";
+import type { SetStateAction } from "react";
+import { useCallback } from "react";
+
+import type { PstComputed } from "../utils/pst-extractor";
+import { computedRoot } from "../utils/pst-extractor";
+
+export interface UsePSTInterface {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/PSTStore.tsx:14]: done

File: src/renderer/store/PSTStore.tsx [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+    mainInfos: PstEmail | undefined;
+    setMainInfos: (update: SetStateAction<PstEmail | undefined>) => void;
+    depth: number | string;
+    setDepth: (update: SetStateAction<number>) => void;
+}
+
+const sentFolderAtom = atom("");
+const deletedFolderAtom = atom("");
+const rootPathAtom = atom("");
+const pstFileAtom = atom<PstContent | undefined>(undefined);
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/PSTStore.tsx:39]: good point ! 
Prefer to use `void 0` so :)

File: src/renderer/store/PSTStore.tsx [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+const computedPstAtom = atom<PstComputed | undefined>(undefined);
+const pstExtractTablesAtom = atom<PstExtractTables | undefined>(undefined);
+const pstMainInfosAtom = atom<PstEmail | undefined>(undefined);
+const pstDepthAtom = atom<number>(0);
+
+// const getterAtom = atom((get) => get(pstDepthAtom));
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/PSTStore.tsx:45]: done

File: src/renderer/store/PSTStore.tsx [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+// const setterAtom = atom((set) => set(pstDepthAtom));
+
+export const usePSTStore = (): UsePSTInterface => {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/store/PSTStore.tsx:48]: done

File: src/renderer/hooks/usePSTFolderList.tsx [hunk 24]
```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+import { usePSTStore } from "../store/PSTStore";
+import { getPSTRootFolderList } from "../utils/pst-extractor";
+
+interface UsePSTExtractor {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/usePSTFolderList.tsx:6]: done

File: src/renderer/hooks/usePSTFolderList.tsx [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+    folderList: string[];
+}
+
+export const usePSTFolderList = (): UsePSTExtractor => {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/usePSTFolderList.tsx:10]: done

File: src/renderer/hooks/usePSTExtractor.tsx [hunk 26]
```diff
@@ -0,0 +1,59 @@
+import { useService } from "@common/modules/ContainerModule";
+import type { PstProgressState } from "@common/modules/pst-extractor/type";
+import type { Dispatch, SetStateAction } from "react";
+import { useEffect, useState } from "react";
+
+import { usePSTStore } from "../store/PSTStore";
+
+interface UsePSTExtractor {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/usePSTExtractor.tsx:8]: done

File: src/renderer/hooks/usePSTExtractor.tsx [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+    elapsed: 0,
+    progress: false,
+};
+
+/**
+ * A hook that launch the extraction of the service and exposes PST extracted file and progress.
+ *
+ * @returns extracted file and progress
+ */
+export const usePSTExtractor = (): UsePSTExtractor => {
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/usePSTExtractor.tsx:27]: done

File: src/renderer/hooks/usePSTExtractor.tsx [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+    const [pstFilePath, setPstFilePath] = useState<string>("");
+    const [pstProgress, setPstProgress] = useState<PstProgressState>(
+        pstProgressInitialState
+    );
+    const pstExtractorService = useService("pstExtractorService");
+
+    const { setPstFile, setExtractTables, updateComputedPst } = usePSTStore();
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/usePSTExtractor.tsx:34]: done

File: src/renderer/hooks/useContextMenu.tsx [hunk 29]
```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { useTagManagerStore } from "../store/TagManagerStore";
+
+interface UseContextMenuType {
+    anchorPoint: {
+        x: number;
+        y: number;
+    };
+    show: boolean | null;
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/useContextMenu.tsx:10]: done

File: src/renderer/hooks/useContextMenu.tsx [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+export const useContextMenu = (): UseContextMenuType => {
+    const [anchorPoint, setAnchorPoint] = useState({ x: 0, y: 0 });
+    const [show, setShow] = useState<boolean | null>(false);
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/useContextMenu.tsx:15]: yes... better than `Pick` or `Omit` in this use case.

File: src/renderer/hooks/useContextMenu.tsx [hunk 31]
```diff
@@ -0,0 +1,64 @@
...
+            setShow(true);
+        },
+        [setShow, setAnchorPoint]
+    );
+
+    const handleClick = useCallback(() => {
+        if (show) {
+            setShow(false);
+        }
+        setShow(null);
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/useContextMenu.tsx:35]: `void 0`:)

File: src/renderer/components/viewer/CirclePackingViewer.tsx [hunk 32]
```diff
@@ -0,0 +1,107 @@
...
+    );
+
+    const emptyMaininfos = useCallback(() => {
+        setMainInfos(undefined);
+    }, [setMainInfos]);
+
+    if (!computedPst) return null;
+
+    const borderColor = isToDeleteFolder(computedPst.id, markedToDelete)
+        ? "red"
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/components/viewer/CirclePackingViewer.tsx:88]: done

File: src/renderer/components/viewer/CirclePacking.tsx [hunk 33]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import { usePSTStore } from "../../store/PSTStore";
+import style from "./CirclePacking.module.scss";
+import { CirclePackingViewer } from "./CirclePackingViewer";
+
+export const CirclePacking: React.FC = () => {
+    const { pstFile } = usePSTStore();
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/components/viewer/CirclePacking.tsx:8]: Yes, definitely agree with that!

File: src/renderer/components/dropzone/Dropzone.module.scss [hunk 34]
```diff
@@ -21,11 +22,12 @@
   h1 {
     margin: 0.5rem 0;
     font-size: 1.25rem;
+    font-weight: 600;
   }
   span {
     font-size: 0.75rem;
     color: #aaa;
-    font-weight: bold;
+    // font-weight: bold;
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/components/dropzone/Dropzone.module.scss:30]: done

File: src/renderer/components/common/layout/Layout.tsx [hunk 35]
```diff
@@ -4,18 +4,22 @@ import style from "./Layout.module.scss";
 
 interface LayoutProps {
     className: string;
-    title: string;
+    title?: string;
 }
 
+const LayoutTitle: React.FC<{ title: string | undefined }> = ({ title }) => (
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/components/common/layout/Layout.tsx:10]: done

File: src/renderer/exporters/CsvExporter.ts [hunk 36]
```diff
@@ -0,0 +1,23 @@
+import { writeFile } from "fs/promises";
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/exporters/CsvExporter.ts:1]: Will be added on the next PR `feat/tag-manager`

File: src/renderer/hooks/useContextMenu.tsx [hunk 37]
```diff
@@ -0,0 +1,64 @@
...
+    }, [show]);
+
+    const menu = document.querySelector("#circle-packing");
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/hooks/useContextMenu.tsx:38]: Ref and ForwardRef a bit annoying to use with hooks... `querySelector` works pretty well with React.

File: src/renderer/utils/exporter.ts [hunk 38]
```diff
@@ -0,0 +1,46 @@
+import type { PstExtractTables } from "@common/modules/pst-extractor/type";
```
[the remark this answers is not in the document]
Comment (mehdilouraoui) [reply src/renderer/utils/exporter.ts:1]: yessss ... same as before, will be added on the next PR : `feat/tag-manager`

**Review by lsagetlethias — APPROVED** [review]

**Pull request merged**